### PR TITLE
Updating __getitem__ and __len__ for containers

### PR DESCRIPTION
**Pull request by dasguptar**

### CODE REVIEW

File: torch/nn/modules/container.py [hunk 1]
```diff
@@ -161,6 +168,8 @@ def __init__(self, parameters=None):
             self += parameters
 
     def __getitem__(self, idx):
+	if idx >= len(self._parameters):
```
Comment (apaszke) [inline torch/nn/modules/container.py]: actually this should be
```python
if not (-len(self._parameters) <= idx < len(self._parameters)):
```

**Review by apaszke — APPROVED** [review]

File: torch/nn/modules/container.py (hunk 1 above)
Comment (dasguptar) [inline torch/nn/modules/container.py]: To account for negative indices? This change should be made for the other classes here as well, right?

**Comment by dasguptar** [comment]
@apaszke, I made the suggested changes. Hope it is ok now.

**Review by apaszke — APPROVED** [review]

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
Thanks!

**Pull request merged**